CONTRIBUTING.md

```diff
@@ -0,0 +1,900 @@
+
+Prefer named function declarations to assigning a function to a declared variable where possible.
+
+```js
+// Good
+setTimeout(function () { doSomething() }, 0)
+
+function foo() {
+	return 1
+}
+
+// Good, uses recursion
+setTimeout(function f() {
+	doSomething()
+	if (shouldRepeat()) setTimeout(f, 0)
+}, 0)
+
+// Bad
+requestAnimationFrame(function foo() { runNext() })
+
+var foo = function () {
+	return 1
+}
+```
+
+### Anonymous functions
+
+Anonymous functions must have a single space between `function` and the arguments and between the arguments and the opening brace.
+
+### Named functions, function declarations
+
+Named functions and function declarations must have no space between the function name and the arguments and a single space between the arguments and the opening brace.
+
+## Blocks and conditionals
+
+### Curly braces
+
+Curly braces are required for blocks if the body takes up more than one line. This goes for functions as well. They are optional if it fits within one line, but don't be afraid to use them where you see fit.
+
+Do put an extra space after the first brace and before the final brace in single-line functions.
+
+```js
+// Good
+if (condition) doSomething()
+
+if (condition) {
+	doSomething()
+}
+
+if (condition) {
+	doSomething()
+} else {
+	doSomethingElse()
+}
+
+if (condition) {
+	doSomething()
+	doSomethingElse()
+}
+
+setTimeout(function () { doSomething() }, 0)
+
+setTimeout(function () {
+	doSomething()
+}, 0)
+
+// Bad
+if (condition) { doSomething(); doSomethingElse() }
+
+if (condition)
+	doSomething()
+
+setTimeout(function () { doSomething(); doSomethingElse() }, 0)
+
+setTimeout(function () {doSomething()}, 0) // Body needs some space
+```
+
+### Function declarations
+
+Function declarations should always take more than one line.
+
+```js
+// Good
+function foo() {
+	return bar
+}
+
+// Bad
+function foo() { return bar }
+```
+
+### `if`-`else`
+
+Be consistent with braces in `if`-`else` statements.
+
+```js
+// Good
+if (condition) doSomething()
+else doSomethingElse()
+
+if (condition) {
+	doSomething()
+} else {
+	doSomethingElse()
+}
+
+if (condition) {
+	doSomething()
+} else if (anotherCondition) {
+	doSomethingElse()
+} else {
+	doAnotherThing()
+}
+
+// Okay
+if (condition) doSomething()
+else if (anotherCondition) doSomethingElse()
+else doAnotherThing()
+
+// Bad
+if (condition) doSomething()
+else {
+	doSomethingElse()
+}
+
+if (condition) doSomething()
+else if (anotherCondition) {
+	doSomethingElse()
+} else doAnotherThing()
+
+
+if (condition) {
+	doSomething()
+} else doSomethingElse()
+```
+
+### `do`-`while` loops
+
+This should be avoided in most cases, since the only good use case is if the intent is "do this, and then check" each iteration. This should only be used in cases like below:
+
+```js
+// Form 1
+doSomething()
+
+while (condition) {
+	doSomething()
+}
+
+// Form 2
+while (true) {
+	doSomething()
+	if (!condition) break
+}
+```
```

Comment:
This section is unnecessary. It ought to be obvious that if a do-while is the best way to express something, it should be preferred, otherwise no.